### PR TITLE
merge tao to master

### PR DESCRIPTION
**Pull request by siyliepfl**

### CODE REVIEW

File: qdtrack/models/mot/quasi_dense.py [hunk 1]
```diff
@@ -8,11 +8,30 @@
...
+        if self.freeze_detector:
+            self._freeze_detector()
+        self.is_tao = is_tao
+
+    def _freeze_detector(self):
+        self.backbone.eval()
+        self.neck.eval()
+        self.rpn_head.eval()
+        self.roi_head.bbox_head.eval()
+        for param in self.backbone.parameters():
```
Comment (XiaLiPKU) [inline qdtrack/models/mot/quasi_dense.py:26]: How about writing the following 8 lines into a for loop?

File: qdtrack/models/mot/quasi_dense.py [hunk 2]
```diff
@@ -87,7 +106,10 @@ def simple_test(self, img, img_metas, rescale=False):
                                   self.roi_head.bbox_head.num_classes)
 
         if track_feats is not None:
-            track_result = track2result(bboxes, labels, ids)
+            if self.is_tao:
+                track_result = track2result(bboxes, labels, ids,self.roi_head.bbox_head.num_classes)
```
Comment (XiaLiPKU) [inline qdtrack/models/mot/quasi_dense.py:110]: A blank is needed after the last comma

File: tools/convert_datasets/bdddet2coco.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import argparse
```
Comment (XiaLiPKU) [inline tools/convert_datasets/bdddet2coco.py:1]: This file has been removed by the PR 34.
To get the update, you can do this:
`git merge master; git push origin tao`

Comment (XiaLiPKU) [reply tools/convert_datasets/bdddet2coco.py:1]: Before push, also check the running

File: tools/convert_datasets/tao2ours.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import argparse
```
Comment (XiaLiPKU) [inline tools/convert_datasets/tao2ours.py:1]: Please name it as `tao2coco.py`

File: qdtrack/datasets/tao_dataset.py [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+from pycocotools.coco import COCO
+
+from .coco_video_dataset import CocoVideoDataset
+from .parsers import CocoVID
+
+
+@DATASETS.register_module()
+class TaoDataset(CocoVideoDataset):
+
+    def __init__(self, *args, **kwargs):
```
Comment (XiaLiPKU) [inline qdtrack/datasets/tao_dataset.py:16]: This function can be removed, as it does nothing here

File: configs/tao/ft_qdtrack_frcnn_r50_fpn_24e_tao.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
+# model settings
```
Comment (XiaLiPKU) [inline configs/tao/ft_qdtrack_frcnn_r50_fpn_24e_tao.py:1]: Give these two config files are quite similar for most items.
I suggest let one inherit another. You can examples in mmdet

File: tools/convert_datasets/bdddet2coco.py (hunk 3 above)
Comment (XiaLiPKU) [reply tools/convert_datasets/bdddet2coco.py:1]: After merging, also add some descriptions at docs/GET_START.md

Comment (siyliepfl) [reply tools/convert_datasets/bdddet2coco.py]: bdd2coco is not relevant to this merge. I just delete those two files.

File: qdtrack/models/mot/quasi_dense.py (hunk 1 above)
Comment (siyliepfl) [reply qdtrack/models/mot/quasi_dense.py]: like this?

File: tools/convert_datasets/tao2ours.py (hunk 4 above)
Comment (siyliepfl) [reply tools/convert_datasets/tao2ours.py]: done!

File: qdtrack/datasets/tao_dataset.py (hunk 5 above)
Comment (XiaLiPKU) [reply qdtrack/datasets/tao_dataset.py:16]: __init__

File: tools/convert_datasets/bdddet2coco.py (hunk 3 above)
Comment (XiaLiPKU) [reply tools/convert_datasets/bdddet2coco.py]: `git pull origin master` is a better option

**Review by XiaLiPKU — APPROVED** [review]

**Pull request merged**